### PR TITLE
first version

### LINKED ISSUE #52: mmrm slides for Pharma RUG in China
create slides based on current materials

### PR DESCRIPTION
**Pull request by clarkliming**
close #52 

the duration of the presentation is 25 minutes. I think we should reduce the ASA BIOP SWE intro part, but focus more on the mmrm package itself.

### CODE REVIEW

File: slides/pharma-rug-mmrm-mar2023.qmd [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+::: nonincremental
+::: columns
+::: {.column width="50%"}
+Authors:
+
+-   Brian Matthew Lang (MSD)
+-   Daniel Sabanes Bove (Roche, lead)
+-   Doug Kelkhoff (Roche)
+-   Julia Dedic (Roche)
+-   Kevin Kunzmann (Boehringer Ingelheim)
```
Comment (danielinteractive) [inline slides/pharma-rug-mmrm-mar2023.qmd:31]: I think we have a few more authors now on GitHub, please check and include them

File: slides/pharma-rug-mmrm-mar2023.qmd [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+## Software Engineering in Biostatistics
+
+-   Open-source software has gained increasing popularity in Biostatistics over the last two decades
+    -   Pros: rapid uptake of novel statistical methods and unprecedented opportunities for collaboration and innovation
+    -   Cons: users face huge variability in software quality, in particular reliability, efficiency and maintainability
+-   Developing high-quality software with good coding practices, reproducible outputs, and self-sufficient documentation is critical to inform clinical and regulatory decisions
+
+## How to deal with these issues?
+
+-   To deal with the issues of statistical quality assurances for R packages and creating high quality statistical software a group came together to create through the "Software Engineering Working Group" (SWE WG) to look R packages from a statistical point of view
```
Comment (danielinteractive) [inline slides/pharma-rug-mmrm-mar2023.qmd:66]: this can shorten a bit and integrate into next slide

File: slides/pharma-rug-mmrm-mar2023.qmd [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+## Agenda
+
+-   Introduction and Overview of SWE WG
+-   Mixed Models for Repeated Measures - Why is it a Problem?
+-   MMRM Package
+    -   Why this is not "yet another package"
+    -   Comparing MMRM R Package to other Implementations
+-   Closing and Next Steps
+
+## Software Engineering in Biostatistics
```
Comment (danielinteractive) [inline slides/pharma-rug-mmrm-mar2023.qmd:57]:
```suggestion
## Context: Software Engineering in Biostatistics
```

File: slides/pharma-rug-mmrm-mar2023.qmd [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+
+-   Open-source software has gained increasing popularity in Biostatistics over the last two decades
+    -   Pros: rapid uptake of novel statistical methods and unprecedented opportunities for collaboration and innovation
+    -   Cons: users face huge variability in software quality, in particular reliability, efficiency and maintainability
+-   Developing high-quality software with good coding practices, reproducible outputs, and self-sufficient documentation is critical to inform clinical and regulatory decisions
```
Comment (danielinteractive) [inline slides/pharma-rug-mmrm-mar2023.qmd:62]: You could shorten like this:
```suggestion
-   Open-source software has gained increasing popularity in Biostatistics
    -   Pros: rapid uptake of novel statistical methods, unprecedented opportunities for collaboration
    -   Cons: variability in software quality
-   Developing high-quality software is critical
```

File: slides/pharma-rug-mmrm-mar2023.qmd [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+
+## How to deal with these issues?
+
+-   To deal with the issues of statistical quality assurances for R packages and creating high quality statistical software a group came together to create through the "Software Engineering Working Group" (SWE WG) to look R packages from a statistical point of view
+
+## Software Engineering working group (SWE WG)
+
+-   An official working group of the ASA Biopharmaceutical Section
+-   Formed in August 2022
+-   Cross-industry collaboration with more than 30 members from over 20 organizations
```
Comment (danielinteractive) [inline slides/pharma-rug-mmrm-mar2023.qmd:72]:
```suggestion
-   Cross-industry collaboration (>35 members from >25 organizations)
```

File: slides/pharma-rug-mmrm-mar2023.qmd [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+-   Fast `C++` framework for defining objective functions (`Rcpp` would have been alternative interface)
+-   Automatic differentiation of the log-likelihood as a function of the variance parameters
+-   Syntactic sugars to allow simple matrix calculations or operations like R
+-   We get the gradient and Hessian exactly and without additional coding
+-   This can be used from the R side with the `TMB` interface and plugged into optimizers
+
+## Why it's not just another package
+
+-   Ongoing maintenance and support from the pharmaceutical industry
+    -   5 companies being involved in the development, on track to become standard package
```
Comment (danielinteractive) [inline slides/pharma-rug-mmrm-mar2023.qmd:115]: check number here, maybe it is already 6 now or so :-)

File: slides/pharma-rug-mmrm-mar2023.qmd [hunk 7]
```diff
@@ -0,0 +1,339 @@
...
+-   REML or ML estimation, using multiple optimizers if needed
+-   `emmeans` interface for least square means
+-   Satterthwaite adjusted degrees of freedom
+-   Kenward-Roger adjusted degrees of freedom
+-   Robust sandwich estimator for covariance
+
+
+## Comparing SAS and R
+
+To run an MMRM model in SAS it is recommended to use either the PROC MIXED or PROC GLM procedures.
```
Comment (danielinteractive) [inline slides/pharma-rug-mmrm-mar2023.qmd:135]:
```suggestion
To run an MMRM model in SAS it is recommended to use either the `PROC MIXED` or `PROC GLM` procedures.
```

Comment (danielinteractive) [reply slides/pharma-rug-mmrm-mar2023.qmd:135]: same below

File: slides/pharma-rug-mmrm-mar2023.qmd [hunk 8]
```diff
@@ -0,0 +1,339 @@
...
+-   [Usage](https://documentation.sas.com/doc/en/pgmsascdc/9.4_3.4/statug/statug_mixed_examples02.htm)
+-   [Theory](https://documentation.sas.com/doc/en/pgmsascdc/9.4_3.4/statug/statug_mixed_details01.htm)
+-   [Covariance Structures](https://documentation.sas.com/doc/en/pgmsascdc/9.4_3.4/statug/statug_mixed_syntax14.htm#statug.mixed.repeatedstmt_type)
+-   [Degrees of Freedom](https://documentation.sas.com/doc/en/pgmsascdc/9.4_3.4/statug/statug_mixed_syntax10.htm#statug.mixed.modelstmt_ddfm)
+:::
+:::
+
+## Unit Testing
+
+One major advantage of the `{mmrm}` over `PROC MIXED` is that the unit testing in `{mmrm}` is transparent. It uses the `{testthat}` framework with `{covr}` to communicate the testing coverage. Unit tests can be found in the GitHub repository under [./tests](https://github.com/openpharma/mmrm/tree/main/tests/testthat).
```
Comment (danielinteractive) [inline slides/pharma-rug-mmrm-mar2023.qmd:189]: maybe make this more a bulleted list? for better readability

File: slides/pharma-rug-mmrm-mar2023.qmd [hunk 9]
```diff
@@ -0,0 +1,339 @@
...
+::: {.column width="60%"}
+-   SAS has 23 non-spatial covariance structures, while mmrm has 10.
+    -   9 structures intersect with SAS
+    -   Ante-dependence (homogeneous) is only in the mmrm package.
+-   SAS has 14 spatial covariance structures compared to the spatial exponential one available in mmrm.
+:::
+
+::: {.column width="40%"}
+::: callout-tip
+For users that need more structure `{mmrm}` is easily extensible via [feature requests](https://github.com/openpharma/mmrm/issues/new?assignees=&labels=enhancement&template=feature_request.md&title=) in the GitHub repository.
```
Comment (danielinteractive) [inline slides/pharma-rug-mmrm-mar2023.qmd:214]:
```suggestion
For users that need more structures, `{mmrm}` is easily extensible - let us know via [feature requests](https://github.com/openpharma/mmrm/issues/new?assignees=&labels=enhancement&template=feature_request.md&title=) in the GitHub repository.
```

File: slides/pharma-rug-mmrm-mar2023.qmd [hunk 10]
```diff
@@ -0,0 +1,339 @@
...
+| Method                     | `{mmrm}` | `PROC MIXED` |
+|:---------------------------|:--------:|:------------:|
+| Contain                    |   X\*    |      X       |
+| Between/Within             |   X\*    |      X       |
+| Residual                   |   X\*    |      X       |
+| Satterthwaite              |    X     |      X       |
+| Kenward-Roger              |    X     |      X       |
+| Kenward-Roger (Linear)\*\* |    X     |      X       |
+
+\*Available through the `emmeans` package.
```
Comment (danielinteractive) [inline slides/pharma-rug-mmrm-mar2023.qmd:253]:
```suggestion
\*Available through the `emmeans` package, and ongoing work in `mmrm`.
```

File: slides/pharma-rug-mmrm-mar2023.qmd [hunk 11]
```diff
@@ -0,0 +1,339 @@
...
+
+\*\*This is not identical to the KR-Linear setting in `PROC MIXED` because of different
+parameterization.
+
+## Contrasts/LSMEANS
+
+Contrasts and LSMeans estimates are available in `mmrm` using
```
Comment (danielinteractive) [inline slides/pharma-rug-mmrm-mar2023.qmd:260]:
```suggestion
Contrasts and least square (LS) means estimates are available in `mmrm` using
```

**Review by danielinteractive — COMMENTED** [review]
thanks Liming looks good, see few suggestions inside. How much time do you have in total at the conference ?

**Comment by clarkliming** [comment]
> thanks Liming looks good, see few suggestions inside. How much time do you have in total at the conference ?

as mentioned in the description of the PR, it is 25 minutes

**Comment by danielinteractive** [comment]
> > thanks Liming looks good, see few suggestions inside. How much time do you have in total at the conference ?
> 
> as mentioned in the description of the PR, it is 25 minutes

Ah ok. I thought you meant that you tested the slides and it took you 25 mins :-D. I would say it is ok for that! No need to cut much shorter. I would just train it once with a timer to make sure you know that it works so you don't get stressed

File: slides/pharma-rug-mmrm-mar2023.qmd (hunk 6 above)
Comment (clarkliming) [reply slides/pharma-rug-mmrm-mar2023.qmd:115]: in the description page, only BI, Gilead, Roche, Merck and AZ

**Review by danielinteractive — APPROVED** [review]

**Pull request merged**